### PR TITLE
WP.com block editor: Exclude WP.com features from Jetpack sites

### PR DESCRIPTION
**Pull request by mmtr**
#### Changes proposed in this Pull Request:

In order to don't expose some `@automattic/wpcom-block-editor` features to Jetpack sites that are specific of WP.com sites, the `common` bundle has been split into two new bundles `default` and `wpcom` in https://github.com/Automattic/wp-calypso/pull/38248.

This PRs changes the WordPress.com block editor module to replace the `common` bundle with the new  `default` and to load `wpcom` only when the site is a WP.com site (Atomic).

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

Changes the existing WordPress.com block editor module.

#### Testing instructions:

* Spin up a new [JN site running this branch](https://jurassic.ninja/create/?jetpack-beta&branch=update/wpcom-block-editor-exclude-features-jetpack).
* Verify the following features still work in both Calypso and WP Admin block editors:
  - Justify and underline format.
* Verify the following features still work only in the Calypso block editor:
  - Switch to classic (from the "More tools" menu).
  - Core media modal replaced by Calypso media modal.
  - Posts are previewed with the Calypso preview modal.
  - Revisions are managed with the Calypso revisions modal.
  - Media images in classic blocks are added with the Calypso media modal.
* Verify the following features are no longer available in both Calypso and WP Admin block editors:
  - Disable welcome guide tour on new sites (users should see the welcome guide the first time they use the block editor).
  - Reorder block categories (if CoBlocks is installed, it should be at the top).
  - Disable experimental blocks (those blocks should be available).
  - Attempt block recovery on editor load (invalid blocks should remain invalid).

#### Proposed changelog entry for your changes:

* WP.com block editor: Exclude WP.com features from Jetpack sites

### CODE REVIEW

File: modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php
Comment (jeherve) [inline modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php]: Don't we need `wp-polyfill` for IE11 support?

File: modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php
Comment (mmtr) [inline modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php]: This is intended for loading the `wpcom.js` script on Atomic sites, but it seems `IS_WPCOM` is not defined on those sites. Do we have any other way of performing such check?

Comment (jeherve) [reply modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php]: We have `jetpack_is_atomic_site()` that you could use to check for Atomic sites.

Comment (mmtr) [reply modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php]: Maybe [`jetpack_is_atomic_site`](https://github.com/Automattic/jetpack/blob/31efa189ad223c0eb7ad085ac0650a23facf9ef5/functions.global.php#L75)

**Review by jeherve — CHANGES_REQUESTED** [review]

**Review by jeherve — DISMISSED** [review]
Other than my question [here](https://github.com/Automattic/jetpack/pull/14230/files#r357714536), this seems to work well in my tests.

I have not tested on Atomic though

File: modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php
Comment (mmtr) [reply modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php]: It's already handled by `Jetpack_Gutenberg`:

https://github.com/Automattic/jetpack/blob/0b8ffe1615f2b8bc09e22e3a5f065f7674a65fdc/class.jetpack-gutenberg.php#L596

**Comment by kwight** [comment]
All tests great for me, wonderful 👍

**Review by kwight — DISMISSED** [review]

**Comment by kwight** [comment]
Oh, hm, I'm not seeing experimental blocks at all (at least, I can't pull up `[GradientPicker](https://github.com/WordPress/gutenberg/blob/f58d0e3024209c63b52912e82454dd6785af8097/packages/block-editor/src/components/index.js#L24)`, for example).

**Comment by mmtr** [comment]
> Oh, hm, I'm not seeing experimental blocks at all (at least, I can't pull up [`GradientPicker`](https://github.com/WordPress/gutenberg/blob/f58d0e3024209c63b52912e82454dd6785af8097/packages/block-editor/src/components/index.js#L24), for example).

That's a component, not a block 🙂 . AFAIK, the only experimental block that is available in Gutenberg right now is the Legacy Widget block. To see it, you need to install the Gutenberg plugin and enable the Widgets experiment:

Screenshot 2019-12-17 at 11 16 34

File: modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php
Comment (gwwar) [inline modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php]: As a follow up we might be able to script out an autogenerated diff later. One tradeoff here for inlining is that we can't update published styles outside of Jetpack release cycles for any other integration changes. Probably okay here

**Review by gwwar — APPROVED** [review]
Tests well for me ✨

File: modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php
Comment (kwight) [reply modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php]: > One tradeoff here for inlining is that we can't update published styles outside of Jetpack release cycles for any other integration changes.

We _could_, by just adding another inline style that overrides what's in Jetpack, until the new JP release comes out – certainly not a fun solution, though 🙃

**Comment by mmtr** [comment]
Just landed D36841-code so this could be finally merged. @jeherve I dismissed your review with my latest changes using the new bundle names, so this needs your approval again.

**Review by jeherve — APPROVED** [review]
Looks good and tests well for me. 🚢

**Pull request merged**
